iam-login-service/src/main/java/it/infn/mw/iam/registration/DefaultRegistrationRequestService.java

```diff
@@ -358,9 +358,12 @@ private RegistrationRequestDto handleConfirm(IamRegistrationRequest request) {
   }
 
   private RegistrationRequestDto handleReject(IamRegistrationRequest request,
-      Optional<String> motivation) {
+      Optional<String> motivation, boolean doNotSendEmail) {
     request.setStatus(REJECTED);
-    notificationFactory.createRequestRejectedMessage(request, motivation);
+    if(!doNotSendEmail){
+      notificationFactory.createRequestRejectedMessage(request, motivation);
+    }
+    
```

Comment:
I would add the `motivation` to the message published into RegistrationRejectEvent (line 371), knowing it's still required

Reply:
Done